docs/fsharp/style-guide/conventions.md

```diff
@@ -527,6 +527,10 @@ It's common to define single-case Discriminated Unions like this for domain mode
 
 Although the previous example showed that a struct Discriminated Union yielded better performance, it is common to have larger Discriminated Unions when modeling a domain. Larger data types like that may not perform as well if they are structs depending on the operations on them, since more copying could be involved.
 
+### Functional programming and nulls
+
+Nulls should normally be avoided, most F# types don't allow nulls, however some .NET types are frequently used if F# and they allow nulls, this is the case for arrays and strings. In practice there's no need to check for nulls but if you do fail fast, this applies to FSharp libraries as well. Although for legacy reasons some string functions in FSharp.Core still treat nulls as empty strings we shouldn't take them as guidance, that is we would not encourage attributing any semantic meaning to "null".
```

Comment:
I really like that we are adding this, and my suggestions here are to add clarity to your points. This is sufficiently important that we can spend more space.

```suggestion
Nulls should generally be avoided in F#. F# supports nulls because some common .NET APIs return or accept nulls and some common types such as arrays and strings allow nulls. However, one of the benefits of using F# is to avoid null reference errors in most cases.

In practice there's no need to check for nulls but if you do fail fast, this applies to FSharp libraries as well. Although for legacy reasons some string functions in FSharp.Core still treat nulls as empty strings we shouldn't take them as guidance, that is we would not encourage attributing any semantic meaning to "null".
```

I am confused by "most F# types don't have nulls". Members? 

Can we clarify this? _In practice there's no need to check for nulls but if you do fail fast, this applies to FSharp libraries as well._ If you are working with .NET APIs, this does not necessarily hold. 

_Although for legacy reasons some string functions in FSharp.Core still treat nulls as empty strings we shouldn't take them as guidance, that is we would not encourage attributing any semantic meaning to "null"._

Reply:
@vzarytovskii

Reply:
from my conversation with @dsyme :
> ... libraries should fail fast for sure
> Should be added to the coding guidelines - "in practice no need to check for nulls but if you do fail fast

Reply:
I'll commit the suggestion and edit a little